packages/core/contracts/chainbridge/SourceGovernor.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "../oracle/interfaces/FinderInterface.sol";
+import "../oracle/implementation/Constants.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./IBridge.sol";
+
+/**
+ * @title Governor contract on L2 that receives governance actions from L1.
+ */
+contract SourceGovernor is Ownable {
+    FinderInterface public finder;
+    uint8 public currentChainId;
+    bytes32 internal currentRequestHash;
```

Comment:
So this means that we can only send one governance action at a time? I think this is a reasonable constraint, but I wonder if there is a reason why you didn't use an array of governance actions.

Reply:
1. That's how the mainnet governor works (i.e. each transaction within a governance proposal must be executed independently). The reason we did this on mainnet was to ensure that you don't end up with block gas limit issues if you try to do too much in a single proposal (like the omnibus UMIP, for instance). 
2. Each transaction on the L2 would line up with one transaction sent by the governor on mainnet, so it's easier to track.
3. Simplicity of encoding/decoding (no arrays of arrays).

I'm definitely open to changing, but this seems simplest to start with. Thoughts?

Reply:
Let's keep it this way for simplicity of reasoning about!